Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,385 @@
+    }
+
+    /**
+     * @param string $queue
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPQueueException
+     */
+    public function subscribe($queue)
+    {
+        if ($this->currentEnvelope) {
+            throw new \RuntimeException('Don\'t be greedy and process the message you already have!');
+        }
+
+        $this->queue = $this->manager->getQueue((string) $queue);
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function unSubscribe()
+    {
+        $this->disconnect();
+        $this->queue = $this->currentEnvelope = null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function ack()
+    {
+        if (!$this->currentEnvelope) {
+            throw new \RuntimeException('You must first receive a message, before acking it');
+        }
+
+        $this->queue->ack($this->currentEnvelope->getDeliveryTag());
+        $this->currentEnvelope = null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function nack()
+    {
+        if (!$this->currentEnvelope) {
+            throw new \RuntimeException('You must first receive a message, before nacking it');
+        }
+
+        $this->queue->nack($this->currentEnvelope->getDeliveryTag(), AMQP_REQUEUE);
+        $this->currentEnvelope = null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function doDestroy()
+    {
+        $this->disconnect();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(QueueMessageInterface $message, $destination = null)
+    {
+        $messageBody = $this->getSerializer()->serialize($message, $this->format);
+        $messageHeaders = new AMQPTable($message->getHeaders());
+        $message = new AMQPMessage($messageBody);
+        $message->set('application_headers', $messageHeaders);
+        $this->manager->publish($message);
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive()
+    {
+        if ($this->currentEnvelope) {
+            throw new \LogicException('AmqpQueueDriver: You have to subscribe before receiving.');
+        }
+        $waited = 0;
+
+        while (null === ($msg = $this->receiveNoWait()) && static::WAIT_TIMEOUT > $waited) {
+            $this->sleep(static::WAIT_PERIOD);
+            $waited += static::WAIT_PERIOD;
+        }
+
+        return $msg;
+    }
+
+    /**
+     * Returns One Serializable object from the queue.
+     *
+     * It requires to subscribe previously to a specific queue
+     *
+     * @throws \Exception
+     *
+     * @return \Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface|null
+     */
+    public function receiveNoWait()
+    {
+        if ($this->currentEnvelope) {
+            throw new \RuntimeException(
+                'AmqpQueueDriver: This driver has a message that was not acknowledged yet. A message must be processed and acknowledged before receiving new messages.'
```

Comment:
Here is mentioning the driver name of ```AmqpQueueDriver``` instead of ```PhpAmqpLibDriver```

Reply:
This function must to be customized for the new driver too. The reason that it's here is because of the parent abstract class too.